### PR TITLE
Fix interceptor issue

### PR DESCRIPTION
**Pull request by dreab8**
it remains the problem with the session SESSION_INTERCEPTOR, in the old code this interceptor is called by EntityManagerimpl#internalGetSession() so not sure about the use in the new code

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/jpa/boot/internal/EntityManagerFactoryBuilderImpl.java [hunk 1]
```diff
@@ -874,7 +874,7 @@ protected void populate(SessionFactoryBuilder sfBuilder, StandardServiceRegistry
 		final StrategySelector strategySelector = ssr.getService( StrategySelector.class );
 
 		// Locate and apply the requested SessionFactory-level interceptor (if one)
-		final Object sessionFactoryInterceptorSetting = configurationValues.remove( org.hibernate.cfg.AvailableSettings.INTERCEPTOR );
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/jpa/boot/internal/EntityManagerFactoryBuilderImpl.java]: Shouldnt this just use:

	String SESSION_SCOPED_INTERCEPTOR = "hibernate.session_factory.session_scoped_interceptor";

?

Comment (dreab8) [inline hibernate-core/src/main/java/org/hibernate/jpa/boot/internal/EntityManagerFactoryBuilderImpl.java]: 3 tests in org.hibernate.jpa.test.ejb3configuration.InterceptorTest use org.hibernate.jpa.AvailableSettings.INTERCEPTOR 

that is INTERCEPTOR = "hibernate.ejb.interceptor"

Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/jpa/boot/internal/EntityManagerFactoryBuilderImpl.java]: Yep, which is deprecated :)

So one solution is to change the code as as you did, but that just means we will need to change this back later.

Another option is to deal with (and report - DeprecationLogger) the deprecation.  I'll take a look at this one.  I'd prefer this approach.

**Comment by sebersole** [comment]
See 8e07dc4cde32c8ac77982f39fa9c7742d99a2f65

**Pull request closed without merging**